src/main/java/com/minecolonies/coremod/sounds/KnightSounds.java

```diff
@@ -0,0 +1,173 @@
+        public static final SoundEvent greeting          = ModSoundEvents.getSoundID("mob.guard.knight.male.greeting");
+        public static final SoundEvent farewell          = ModSoundEvents.getSoundID("mob.guard.knight.male.farewell");
+        public static final SoundEvent interaction       = ModSoundEvents.getSoundID("mob.guard.knight.male.interaction");
+
+        /**
+         * Private constructor to hide the implicit public one.
+         */
+        private Male()
+        {
+            /*
+             * Intentionally left empty.
+             */
+        }
+    }
+
+    /**
+     * Plays Knight sounds.
+     * Suppressing Sonar Rule squid:S109
+     * This rule wants to prevent magic numbers
+     * But in this case the rule does not apply because its not a magic number its a % chance.
+     * For every sound category we have 1 increasing number.
+     *
+     * @param worldIn    the world to play the sound in.
+     * @param position   the position to play the sound at.
+     * @param isFemale   the gender.
+     * @param saturation the saturation.
+     */
+    @SuppressWarnings("squid:S109")
+    @Override
+    public void playSound(final World worldIn, final BlockPos position, final boolean isFemale, final double saturation)
+    {
+        //While there are no male sounds
+        if (isFemale)
+        {
+            return;
+        }
+        //Leaving it as switch-case we may add further random sound categories here (Whistling, singing, etc).
+        switch (rand.nextInt(NUMBER_OF_SOUNDS + 1))
+        {
+            case 1:
+                final SoundEvent noises = Male.noises;
+                SoundUtils.playSoundAtCitizenWithChance(worldIn, position, noises, getBasicSoundChance());
+                break;
+            case 2:
+                playSaturationSound(worldIn, position, isFemale, saturation);
+                break;
+            case 3:
+                final SoundEvent greeting = Male.greeting;
+                SoundUtils.playSoundAtCitizenWithChance(worldIn, position, greeting, getBasicSoundChance() * 2);
+                break;
+            case 4:
+                final SoundEvent farewell = Male.farewell;
+                SoundUtils.playSoundAtCitizenWithChance(worldIn, position, farewell, getBasicSoundChance());
+                break;
+            default:
+                final SoundEvent generalPhrases = Male.generalPhrases;
+                SoundUtils.playSoundAtCitizenWithChance(worldIn, position, generalPhrases, getPhraseChance());
+                break;
+        }
+    }
+
+    @Override
+    public String getWorkerString()
+    {
+        return "knight";
+    }
+
+    /**
+     * Play interaction sound.
+     *
+     * @param worldIn  world to play it in.
+     * @param position position to play it at.
+     * @param isFemale the gender.
+     */
+    public void playInteractionSound(final World worldIn, final BlockPos position, final boolean isFemale)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "isFemale" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2301)